From 6c6a4a1f91d36defe8f467b45c5110bdcf65e83a Mon Sep 17 00:00:00 2001
From: svenate <salini.venate@intel.com>
Date: Wed, 16 Mar 2022 15:17:35 +0530
Subject: [PATCH] Fix clang 14 error build error

Clang-14 was throwing out below build errror:

hardware/intel/kernelflinger/libkernelflinger/qsort.c:87:7:
error: performing pointer subtraction with a null pointer
has undefined behavior [-Werror,-Wnull-pointer-subtraction]
loop:   SWAPINIT(a, es);

Signed-off-by: svenate <salini.venate@intel.com>
---
 libkernelflinger/qsort.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libkernelflinger/qsort.c b/libkernelflinger/qsort.c
index 7664d58..0f58b5f 100644
--- a/libkernelflinger/qsort.c
+++ b/libkernelflinger/qsort.c
@@ -46,7 +46,7 @@ static __inline void swapfunc(char *, char *, size_t, int);
 	} while (--i > 0);		\
 }
 
-#define SWAPINIT(a, es) swaptype = ((char *)a - (char *)0) % sizeof(long) || \
+#define SWAPINIT(a, es) swaptype = ((uintptr_t)a) % sizeof(long) || \
 	es % sizeof(long) ? 2 : es == sizeof(long)? 0 : 1;
 
 static __inline void
-- 
2.17.1

